frontend/model/captureLogs.js

```diff
@@ -0,0 +1,218 @@
+import sbp from '~/shared/sbp.js'
+import { CAPTURED_LOGS, SET_APP_LOGS_FILTER } from '~/frontend/utils/events.js'
+
+/*
+  - giConsole/count - total logs stored
+  - giConsole/lastEntry - latest log hash
+  - giConsole/limit - the limit of entries.
+  - giConsole/markers - an array of markers at every nth entry
+  - giConsole/markerNth - the nth index in which marker log
```

Comment:
So while testing this while multiple-user accounts across different Firefox Container tabs and the same container tabs here's what I learned:

- Different container tabs have different localStorage instances
- The same container tab (i.e. what most users will use), share the same localStorage instance

So for example, for both users, this console query returned the same value for everything except `'giConsole/lastEntry'`: `['giConsole/count', 'giConsole/lastEntry', 'giConsole/limit', 'giConsole/markers', 'giConsole/markerNth'].map(e => ({[e]: localStorage.getItem(e)}))`

What this means is that the logs are lost between logins into different accounts... and they are even lost (for a different reason) between different logins/logouts of the *same account!*

Give it a try with 80/10 (80 log limit, 10nth). Login and logout. You'll see that each time the saved logs get wiped.

So there are two problems to solve here:

1. Make sure we store different logs for different users. I think the way to do this is to use the username as part of the key (e.g. `giConsole/${username}/count`, `giConsole/${username}/lastEntry ` etc.). Verify this actually works.
2. Save the logs between sessions. I actually don't know why this is happening with the same user. But give a try yourself using 80/10, and I think you'll see that when you logout and log back in a fresh set of logs is stored and the previous logs are gone...

Too late right now for me to dig further into this.

Reply:
> Give it a try with 80/10 (80 log limit, 10nth). Login and logout. You'll see that each time the saved logs get wiped.

Hum... that bug didn't happen on my side. I logged in as "sandy", logged out, logged in again and the logs are still there. Even if I create a 2nd account, "greg", all logs were kept as expected. I tested on Chrome and Firefox Dev Edition (both always using the same tab).  [See screenshot](https://user-images.githubusercontent.com/14869087/77831440-1187d800-7127-11ea-9024-17a31091b274.png). Maybe you could explain the exact steps to reproduce the bug. Make sure to reset the LS before you start. Maybe you had some artifacts from old commits.

But the true question is: Do we really want to keep the logs after the user logs out? Imagine I'll log in to a public computer (a library). Then, I'll log out. The logs will still be there. With all the sensitive data about the user and group info. That's a concern.

> Make sure we store different logs for different users.

Done, even having the security concern mentioned before. And another question: How we handle the limit based on the nº of users? If there are 2 users, maybe the limit should be cut by half? (2500 logs for each user).

In summary: I think for now we should store the logs only while the user is logged in and clean them when they log out. For simplicity and for security reasons.

-----

I also updated the code to avoid infinite loops at `while (entriesCount >= ENTRIES_LIMIT) {...}`. It may happen if for some unknown reason the chain of entries gets broken and we aren't able to delete the oldest logs. See lines 99-109.

Reply:
> But the true question is: Do we really want to keep the logs after the user logs out? Imagine I'll log in to a public computer (a library). Then, I'll log out. The logs will still be there. With all the sensitive data about the user and group info. That's a concern.

Very valid concern. My plan was (still is) to encrypt the logs. But, the idea of additionally wiping them after logging out does sound reasonable to me too. We could do both. Wiping them on logout would need to be explicitly implemented in this PR though, as I don't think it currently does that.

> Done, even having the security concern mentioned before. And another question: How we handle the limit based on the nº of users? If there are 2 users, maybe the limit should be cut by half? (2500 logs for each user).

We can keep the limit as it is, but the addition of wiping the logs on logout would address this as well. Good idea 👍 .

Reply:
Okay, I'll update the code to remove all logs on logout and keep the limit as it is.

Should we keep the logs under /[username]  namespace or should that be reverted? I would revert to keep things simple (only need to change 3 lines)

Reply:
I think it's fine to keep ti under the /[username] namespace 👍